### PR TITLE
[docs] Update FAQ with network troubleshooting information

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**
@AndreyBelym @kirovboris

### CODE REVIEW

File: docs/articles/faq/README.md [hunk 1]
```diff
@@ -259,4 +262,27 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
 to control the browser window size. If the test passes, it means your webpage hides
 the target element when the window is resized to smaller dimensions.
 
-Finally, try updating TestCafe to the latest version to see if the problem persists.
\ No newline at end of file
+Finally, try updating TestCafe to the latest version to see if the problem persists.
+
+### What are the possible reasons for an HTTP request failure? How can I address them?
```
Comment (kirovboris) [inline docs/articles/faq/README.md]: It's not only about HTTP, right?

File: docs/articles/faq/README.md [hunk 2]
```diff
@@ -259,4 +262,27 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+
+#### Unstable or improperly configured network connection
+
+* Check the proxy server's, router's and network connection's settings.
+* Connect to a different network.
+* Restart or replace the router.
+* Use VPN.
+* Bypass or change the proxy server.
+
+#### Not enough resources in the container or CI system
```
Comment (kirovboris) [inline docs/articles/faq/README.md]: IMO, We described here what you can try to find a reason, not how to address this problem.

File: docs/articles/faq/README.md [hunk 3]
```diff
@@ -259,4 +262,27 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+
+* Decrease the [concurrency factor](../documentation/using-testcafe/common-concepts/concurrent-test-execution.md) to `1`.
+* Deploy the application's Web server on a separate machine.
+* Change the container's resource limits.
+* Run tests on a local device outside a container.
+* In a CI service, switch to a pricing plan that offers more resources.
```
Comment (kirovboris) [inline docs/articles/faq/README.md]: It's too imperatively, why should I use TestCafe if I have to change plan?
Even if should, let's say it softer 😄 .

File: docs/articles/faq/README.md [hunk 4]
```diff
@@ -259,4 +262,27 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+* Try a different CI service.
```
Comment (kirovboris) [inline docs/articles/faq/README.md]: Which problem does it solve?

File: docs/articles/faq/README.md [hunk 5]
```diff
@@ -259,4 +262,27 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+
+#### The Web server is not responding
+
+Check if the Web and DNS servers are online and configured to accept requests to this URL.
+
+#### Unstable or improperly configured network connection
+
+* Check the proxy server's, router's and network connection's settings.
+* Connect to a different network.
+* Restart or replace the router.
```
Comment (kirovboris) [inline docs/articles/faq/README.md]: So should I go and buy another router to check this hypothesis?

File: docs/articles/faq/README.md (hunk 2 above)
Comment (kirovboris) [reply docs/articles/faq/README.md]: Everything here is about "you could try", so I'd rephrase it.

File: docs/articles/faq/README.md (hunk 4 above)
Comment (AndreyBelym) [reply docs/articles/faq/README.md]: It can solve the problem with hardware configuration/resources. You can try to find a CI provider that offers better hardware, or a provider that is less popular and don't experience performance degradations due to the high amount of users.

File: docs/articles/faq/README.md (hunk 5 above)
Comment (AndreyBelym) [reply docs/articles/faq/README.md]: Or you can connect your machine to the network directly.

File: docs/articles/faq/README.md (hunk 4 above)
Comment (kirovboris) [reply docs/articles/faq/README.md]: Sure, I just want to say it's not obvious for customer.

File: docs/articles/faq/README.md (hunk 5 above)
Comment (kirovboris) [reply docs/articles/faq/README.md]: Probably we can say more common like "Check your network devices/settings".

**Comment by VasilyStrelyaev** [comment]
It's WIP, but is it closer to what we need now?

File: docs/articles/faq/README.md [hunk 6]
```diff
@@ -264,25 +264,44 @@ the target element when the window is resized to smaller dimensions.
...
-* Change the container's resource limits.
 * Run tests on a local device outside a container.
-* In a CI service, switch to a pricing plan that offers more resources.
-* Try a different CI service.
\ No newline at end of file
+
+If this fixes the tests, it indicates that they require additional resources. You can address this in the following ways:
+
+* Adjust the container's or environment's settings to allocate more resources.
+* If you use a commercial CI system, ask the service provider for an upgrade or consider a different CI service with better hardware or smaller loads.
```
Comment (kirovboris) [inline docs/articles/faq/README.md]: Much better overall. It'd be nice if we can recommend several CI services which most suitable for TestCafe tests.

Comment (kirovboris) [reply docs/articles/faq/README.md]: @AndreyBelym what do you think?

Comment (AndreyBelym) [reply docs/articles/faq/README.md]: I think we can reference CI systems that have a corresponding recipe in our documentation.

(list ranking is quite subjective)

[Azure Pipelines](https://devexpress.github.io/testcafe/documentation/continuous-integration/azure-devops.html)
[GitLab CI](https://devexpress.github.io/testcafe/documentation/continuous-integration/gitlab.html)
[TravisCI](https://devexpress.github.io/testcafe/documentation/continuous-integration/travis.html)
[CircleCI](https://devexpress.github.io/testcafe/documentation/continuous-integration/circleci.html)
[AppVeyor](https://devexpress.github.io/testcafe/documentation/continuous-integration/appveyor.html)

[BitBucket](https://devexpress.github.io/testcafe/documentation/continuous-integration/bitbucket-pipelines.html) (I have no experience with it)

Comment (kirovboris) [reply docs/articles/faq/README.md]: Of course we should not advise systems we have no experience with.

**Comment by VasilyStrelyaev** [comment]
I guess I'm sending it for correction now.

File: docs/articles/faq/README.md [hunk 7]
```diff
@@ -108,9 +109,11 @@ test('test', async t => {
 
 ### How do I work with configuration files and environment variables?
 
-TestCafe works without any configuration.
-It does not have any config files where you can place custom variables.
-However, you can introduce your own configuration file and import it to the test code.
+TestCafe allows you to specify its settings in a [configuration file](../documentation/using-testcafe/configuration-file.md).
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
TestCafe allows you to specify settings in a [configuration file](../documentation/using-testcafe/configuration-file.md).
```

File: docs/articles/faq/README.md [hunk 8]
```diff
@@ -108,9 +109,11 @@ test('test', async t => {
...
+
+If you need to use custom proparties in the configuration, you can create a separate configuration file and import it to the tests.
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
If you need to use custom properties in the configuration, create a separate configuration file and import it to the tests.
```

File: docs/articles/faq/README.md [hunk 9]
```diff
@@ -108,9 +109,11 @@ test('test', async t => {
...
+
+> Vote for the following GitHub issue if you want us to support custom properties in `.testcaferc.json`: [#3593](https://github.com/DevExpress/testcafe/issues/3593)
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]: "Vote"?

File: docs/articles/faq/README.md [hunk 10]
```diff
@@ -108,9 +109,11 @@ test('test', async t => {
...
 
 For example, you need to pass a website's base URL to test code. In this instance, you can create the following `config.json` file:
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
For example, you can create the following `config.json` file to pass a website's base URL to test code:
```

File: docs/articles/faq/README.md [hunk 11]
```diff
@@ -120,7 +123,7 @@ For example, you need to pass a website's base URL to test code. In this instanc
 }
 ```
 
-In the test code, import it as you would do with a regular JavaScript module.
+In the test code, import it as you would do with a regular JavaScript module:
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
In the test code, import it like a regular JavaScript module:
```

File: docs/articles/faq/README.md [hunk 12]
```diff
@@ -232,7 +235,7 @@ npm install -g {pluginName}
 This happens because either:
 
 * one of the [selectors](../documentation/test-api/selecting-page-elements/selectors/README.md) you used in test code does not match any DOM element, or
-* you have tried to specify an [action's target element](https://devexpress.github.io/testcafe/documentation/test-api/actions/#selecting-target-elements) using a wrong CSS selector or a client-side function that returns no element.
+* you have tried to specify an [action's target element](../documentation/test-api/actions/README.md#selecting-target-elements) using a wrong CSS selector or a client-side function that returns no element.
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
* you used a incorrect CSS selector or a client-side function that returns no element to specify an [action's target element](../documentation/test-api/actions/README.md#selecting-target-elements).
```

File: docs/articles/faq/README.md [hunk 13]
```diff
@@ -259,4 +262,54 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+
+* Check the network connection's settings.
+* Ensure that your network equipment works properly. If possible, try to establish direct connection to the Internet/Web server.
+* Check the proxy server's settings or try a different proxy server.
+* Use VPN.
+* Connect to a different network.
+
+#### Not enough resources in the container or CI system
+
+If you run TestCafe in a container or CI system, try the following steps to diagnose resource shortage:
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
If you run TestCafe in a container or CI system, use the following steps to diagnose resource shortage:
```

File: docs/articles/faq/README.md [hunk 14]
```diff
@@ -259,4 +262,54 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+
+The following situations lead to this error most frequently:
+
+#### The Web server is not responding
+
+Check if the Web and DNS servers are online and configured to accept requests to this URL.
+
+#### Unstable or improperly configured network connection
+
+To fix network issues, try the following:
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]: "To fix network issues, try the following:" Is it necessary to add this? The heading is sufficient.

File: docs/articles/faq/README.md [hunk 15]
```diff
@@ -259,4 +262,54 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+* Set the [concurrency factor](../documentation/using-testcafe/common-concepts/concurrent-test-execution.md) to `1`.
+* Deploy the application's Web server on a separate machine.
+* Run tests on a local device outside a container.
+
+If this fixes the tests, it indicates that they require additional resources. You can address this in the following ways:
+
+* Adjust the container's or environment's settings to allocate more resources.
+* If you use a commercial CI system, ask the service provider for an upgrade or consider a different CI service with better hardware or smaller loads.
+
+TestCafe users have positive experience with the following CI systems:
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]: "TestCafe users have positive experience with the following CI systems" and I'm not sure about the point of this sentence

Comment (dirk-pieterse) [reply docs/articles/faq/README.md]: Why is it necessary to say "have positive experience"? and do you mean 'a' positive experience?

File: docs/articles/faq/README.md [hunk 16]
```diff
@@ -259,4 +262,54 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+#### The Web server is not responding
+
+Check if the Web and DNS servers are online and configured to accept requests to this URL.
+
+#### Unstable or improperly configured network connection
+
+To fix network issues, try the following:
+
+* Check the network connection's settings.
+* Ensure that your network equipment works properly. If possible, try to establish direct connection to the Internet/Web server.
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
* Ensure that your network equipment works properly. If possible, establish direct connection to the Internet/Web server.
```

File: docs/articles/faq/README.md [hunk 17]
```diff
@@ -259,4 +262,54 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+
+```text
+A request to https://www.example.com has failed. Use quarantine mode to perform additional attempts to execute this test.
+```
+
+[Quarantine mode](../documentation/using-testcafe/command-line-interface.md#-q---quarantine-mode) helps you complete the tests if the problem is occasional.
+
+However, we recommend that you determine the cause of this issue and address it.
+
+The following situations lead to this error most frequently:
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]: "most frequently"?

File: docs/articles/faq/README.md [hunk 18]
```diff
@@ -259,4 +262,54 @@ and [t.resizeWindow](../documentation/test-api/actions/resize-window.md#setting-
...
+
+### TestCafe reports that a request has failed. What are the possible reasons?
+
+When TestCafe does not receive a successful response from a server, it outputs the following error:
+
+```text
+A request to https://www.example.com has failed. Use quarantine mode to perform additional attempts to execute this test.
+```
+
+[Quarantine mode](../documentation/using-testcafe/command-line-interface.md#-q---quarantine-mode) helps you complete the tests if the problem is occasional.
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
You can [Quarantine mode](../documentation/using-testcafe/command-line-interface.md#-q---quarantine-mode) to complete the tests.
```

File: docs/articles/faq/README.md [hunk 19]
```diff
@@ -247,10 +250,10 @@ After this, use the browser's development tools to check that:
...
   it can be changed using the
-  [`offsetX` and `offsetY`](https://devexpress.github.io/testcafe/documentation/test-api/actions/action-options.html#mouse-action-options)
+  [`offsetX` and `offsetY`](../documentation/test-api/actions/action-options.md#mouse-action-options)
   parameters);
 * the element is not in an `<iframe>` (if it is, use the
-  [t.switchToIframe](https://devexpress.github.io/testcafe/documentation/test-api/working-with-iframes.html) method
+  [t.switchToIframe](../documentation/test-api/working-with-iframes.md) method
   to switch to the appropriate `<iframe>`).
 
 Also, try running the test at full screen.
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]:
```suggestion
Also, try running the test in full screen.
```

File: docs/articles/faq/README.md [hunk 20]
```diff
@@ -247,10 +250,10 @@ After this, use the browser's development tools to check that:
   `visibility` set to `hidden` or the zero `width` or `height`);
 * the element's part targeted by the action is visible (the center of the element by default;
```
Comment (dirk-pieterse) [inline docs/articles/faq/README.md]: Do not use 'by default' and avoid 'using'
